java/dataset/src/main/cpp/CMakeLists.txt

```diff
@@ -63,3 +63,5 @@ add_arrow_test(dataset_jni_test
                jni_util.cc
                EXTRA_INCLUDES
                ${JNI_INCLUDE_DIRS})
+
+install(FILES jni_util.h DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/jni/dataset")
```

Comment:
Sorry for commenting this after we merge this.

Should we use `arrow/jni/dataset` or something? It seems that `jni/dataset` is too generic path.

Reply:
Agh, sorry for not being careful. I think it would have been `arrow/jni/dataset` originally right?

Reply:
Oh, hmm. Indeed, we originally had it under `cpp/src/jni/dataset/jni_util.h`. On the other hand if I peek at a Conda package it's not there (presumably because we don't enable JNI). Perhaps we should make the breaking change and revert this so that downstream packages cannot depend on this accidentally?

@jinchengchenghh, what is the use case for depending on `jni_util.h`?

Reply:
Ok from what I see above, there are some Arrow utilities that are being used. IMO, they should get vendored into the downstream project.

Reply:
We could perhaps keep it for 9.0.0 and explicitly break it for 10.0.0 (these should also get refactored into explicit `internal` namespace and header)

Reply:
In any case, there is some ongoing work on the JNI layer to refactor it, base it on the C Data Interface, and try to minimize the amount of custom (and poorly-tested) code. So these are likely to break again.